### PR TITLE
Squish username before validation

### LINKED ISSUE #10101: Trim whitespace in username
### Pitch

The username field during registration and login should allow a whitespace **postfix**.

This is important because many mobile input methods (on-screen keyboards) insert
a whitespace each time you complete a word, so sometimes a whitespace sneaks in
where it shouldn't (after your username).

The server could easily treat the string as if it didn't have a whitespace postfix.
Any user is almost guaranteed not to have meant anything by it. Whitespace is even
explicitly disallowed in usernames.

The thing is that after your keyboard has
automatically added a whitespace and you left the input field, you can't even
see that whitespace, so you will just get confused why mastodon isn't accepting
your new username.

### PR DESCRIPTION
**Pull request by aurelien-reeves**
Fix #10101

### CODE REVIEW

File: app/models/account.rb [hunk 1]
```diff
@@ -78,6 +78,10 @@ class Account < ApplicationRecord
   validates :note, note_length: { maximum: 160 }, if: -> { local? && will_save_change_to_note? }
   validates :fields, length: { maximum: 4 }, if: -> { local? && will_save_change_to_fields? }
 
+  before_validation(on: :create) do
```
Comment (Gargron) [inline app/models/account.rb]: Style-wise, it would be preferable to do it via a private method

`before_validation :prepare_username, on: :create`

I am also not sure what `squish` gives here over `strip`, would prefer to just use `strip`

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/models/account.rb (hunk 1 above)
Comment (aurelien-reeves) [reply app/models/account.rb]: Indeed, I will move `before_validation` into a private method.

Regarding `squish`, actually it processes unicode whitespaces. `strip` does not.

```ruby
pry(main)> "\u3000test\u00a0".strip
=> " test "
pry(main)> "\u3000test\u00a0".squish
=> "test"
```

If you agree to use `squish` to process Unicode white-spaces, I will add a dedicated test for that, or at least I will add Unicode white-spaces into the test I have added.

**Review by Gargron — APPROVED** [review]
